example_test.go

```diff
@@ -0,0 +1,60 @@
+package parquet_test
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/segmentio/parquet-go"
+)
+
+func Example() {
+	// parquet-go uses the same struct-tag definition style as JSON and XML
+	type Contact struct {
+		Name string `parquet:"name"`
+		// "zstd" specifies the compression for this column
+		PhoneNumber string `parquet:"phoneNumber,optional,zstd"`
+	}
+
+	type AddressBook struct {
+		Owner             string    `parquet:"owner,zstd"`
+		OwnerPhoneNumbers []string  `parquet:"ownerPhoneNumbers,gzip"`
+		Contacts          []Contact `parquet:"contacts"`
+	}
+
+	f, _ := os.Create("testdata/data.parquet")
+	writer := parquet.NewWriter(f)
+	rows := []AddressBook{
+		{Owner: "UserA", Contacts: []Contact{
+			{Name: "Alice", PhoneNumber: "+15505551234"},
+			{Name: "Bob"},
+		}},
+		// Add more rows here.
+	}
+	for _, row := range rows {
+		if err := writer.Write(row); err != nil {
+			log.Fatal(err)
+		}
+	}
+	_ = writer.Close()
```

Comment:
Error handling seems to be a bit inconsistent here, we check on `Write` but not on `Close`, which might error since it writes the file footer etc... Maybe the example encourages the wrong practice here, how about checking the error returned by `Close`?

Reply:
Yeah sorry was trying to strike a balance between brevity and explanation

Reply:
My opinion is we should do the error check, younger engineers will use these examples as "good practice" without really thinking through the consequences of not checking the errors, we will serve them better by not teaching them to ignore errors I think.